R/git.R

```diff
@@ -132,6 +132,102 @@ use_git_config <- function(scope = c("user", "project"), ...) {
   }
 }
 
+#' Choose git protocol and credential defaults
+#'
+#' Decide whether to use the SSH or HTTPS protocol for the current session.
+#' This affects two things:
+#'   * The URL format of newly configured git remotes:
+#'     - `protocol = "ssh"` implies `git@@github.com:<OWNER>/<REPO>.git`
+#'     - `protocol = "https"` implies `https://github.com/<OWNER>/<REPO>.git`
+#'   * The strategy for creating default credentials when none are given.
```

Comment:
Is this true? Why aren't we using the same protocol as the repository?

Reply:
I'm just trying to explain the significance of the git protocol, i.e. how it affects usethis's actions.

This is getting reworked as I add similar functionality for `credentials`. So you will get a proper chance to review that PR.